src/types/privatekubectl.ts

```
@@ -0,0 +1,143 @@
+         kubectlCmd
+      ]
+
+      if (addFileFlag) {
+         const filenames = this.extractFilesnames(kubectlCmd).split(' ')
+         const tempDirectory =
+            process.env['runner.tempDirectory'] || os.tmpdir() + '/manifests'
+         eo.cwd = tempDirectory
+         core.debug('ExecOptions current working directory: ' + eo.cwd)
+         privateClusterArgs.push(...['--file', '.'])
+
+         let filenamesArr = filenames[0].split(',')
+         for (let index = 0; index < filenamesArr.length; index++) {
+            var file = filenamesArr[index]
+
+            if (!file) {
+               continue
+            }
+            core.debug('Attempting to move file: ' + file)
+            this.moveFileToTempManifestDir(file)
+         }
+      }
+
+      core.debug(
+         `private cluster Kubectl run with invoke command: ${kubectlCmd}`
+      )
+      core.debug('EO as it goes into getExec ' + eo.cwd)
+      return await getExecOutput('az', privateClusterArgs, eo)
+   }
+
+   public extractFilesnames(strToParse: string) {
+      console.log('string to parse extractFiles: ' + strToParse)
+      var start = strToParse.indexOf('-filename')
+      var offset = 7
+
+      if (start == -1) {
+         start = strToParse.indexOf('-f')
+
+         if (start == -1) {
+            return ''
+         }
+         offset = 0
+      }
+
+      var temp = strToParse.substring(start + offset)
+      var end = temp.indexOf(' -')
+
+      //End could be case where the -f flag was last, or -f is followed by some additonal flag and it's arguments
+      return temp.substring(3, end == -1 ? temp.length : end).trim()
+   }
+
+   private containsFilenames(str: string) {
+      return str.includes('-f ') || str.includes('filename ')
+   }
+
+   private createTempManifestsDirectory() {
+      if (!fs.existsSync('/tmp/manifests')) {
+         try {
+            fs.mkdirSync('/tmp/manifests', {recursive: true})
+         } catch (e) {
+            core.debug(
```

we should just throw an error if we can't make temp. Not a state we can recover from?